presentation/common/src/presentation-common/content/PropertyValueFormatter.ts

```diff
@@ -120,76 +120,89 @@ export class ContentPropertyValueFormatter {
+  private async formatPrimitiveValue(field: Field, value: Value, ctx?: { doubleFormatter: (raw: number) => Promise<string> }) {
     if (value === undefined) {
       return "";
     }
 
     const formatDoubleValue = async (raw: number) => (ctx ? ctx.doubleFormatter(raw) : formatDouble(raw));
 
-    if (type.typeName === "point2d" && isPoint2d(value)) {
+    if (field.type.typeName === "point2d" && isPoint2d(value)) {
       return `X: ${await formatDoubleValue(value.x)}; Y: ${await formatDoubleValue(value.y)}`;
     }
-    if (type.typeName === "point3d" && isPoint3d(value)) {
+    if (field.type.typeName === "point3d" && isPoint3d(value)) {
       return `X: ${await formatDoubleValue(value.x)}; Y: ${await formatDoubleValue(value.y)}; Z: ${await formatDoubleValue(value.z)}`;
     }
-    if (type.typeName === "dateTime") {
+    if (field.type.typeName === "dateTime") {
       assert(typeof value === "string");
       return value;
     }
-    if (type.typeName === "bool" || type.typeName === "boolean") {
+    if (field.type.typeName === "bool" || field.type.typeName === "boolean") {
       assert(typeof value === "boolean");
       return value ? "@Presentation:value.true@" : "@Presentation:value.false@";
     }
-    if (type.typeName === "int" || type.typeName === "long") {
+    if (field.type.typeName === "int" || field.type.typeName === "long") {
       assert(isNumber(value));
       return value.toFixed(0);
     }
-    if (type.typeName === "double") {
+    if (field.type.typeName === "double") {
       assert(isNumber(value));
       return formatDoubleValue(value);
     }
-    if (type.typeName === "navigation") {
+    if (field.type.typeName === "navigation") {
       assert(Value.isNavigationValue(value));
       return value.label.displayValue;
     }
 
+    if (field.type.typeName === "enum" && field.isPropertiesField()) {
+      return field.properties[0].property.enumerationInfo?.choices.find(({ value: enumValue }) => enumValue === value)?.label;
+    }
     // eslint-disable-next-line @typescript-eslint/no-base-to-string
     return value.toString();
   }
 
-  private async formatStructValue(type: StructTypeDescription, value: Value) {
-    if (!Value.isMap(value)) {
+  private async formatStructValue(type: StructTypeDescription, value: Value, field: Field) {
+    if (!Value.isMap(value) || !field.isPropertiesField() || !field.isStructPropertiesField()) {
       return {};
     }
 
     const formattedMember: DisplayValuesMap = {};
     for (const member of type.members) {
-      formattedMember[member.name] = await this.formatValue(member.type, value[member.name]);
+      const fieldMember = field.memberFields.find((memberField) => {
+        return memberField.name === member.name;
+      });
+
+      if (fieldMember !== undefined) {
+        formattedMember[member.name] = await this.formatValue(fieldMember, value[member.name]);
+      }
     }
```

Comment:
Looks like we could iterate over `field.memberFields` and in that case we wouldn't even need the `type` argument